### PR TITLE
fixes #8481 - adds content override to content host API, BZ1166891

### PR DESCRIPTION
**Pull request by cfouant**

### CODE REVIEW

**Comment by cfouant** [comment]
https://github.com/Katello/hammer-cli-katello/pull/304 depends upon this p-r

File: app/models/katello/system.rb [hunk 1]
```diff
@@ -364,6 +364,10 @@ def import_applicability(partial = true)
       end
     end
 
+    def available_content
+      self.products.map(&:available_content).flatten
```
Comment (daviddavis) [inline app/models/katello/system.rb]: You could consider using `flat_map` here.

File: app/controllers/katello/api/v2/systems_controller.rb [hunk 2]
```diff
@@ -254,11 +255,35 @@ def releases
...
+        :results => content,
+        :total => content.size,
+        :subtotal => content.size
+      }
+      respond_for_index :collection => response
+    end
+
+    api :GET, "/systems/:id/content_override", N_("Get content overrides for the content host")
+    param :id, String, :desc => N_("UUID of the content host"), :required => true
+    def product_content
```
Comment (daviddavis) [inline app/controllers/katello/api/v2/systems_controller.rb]: Might be nice to have a test for this action, and content_override and/or validate_content_overrides.

File: app/views/katello/api/v2/systems/product_content.json.rabl [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  end
+
+  child :content => :content do
+    attributes :id, :name, :label
+  end
+
+  node :override do |pc|
+    pc.content_override(@system)
+  end
+end
```
Comment (daviddavis) [inline app/views/katello/api/v2/systems/product_content.json.rabl]: Do we need two separate templates? Looks like the code is the same as `content_override.json.rabl`.

File: config/routes/api/v2.rb [hunk 4]
```diff
@@ -223,6 +223,7 @@ class ActionDispatch::Routing::Mapper
             get :releases
             put :refresh_subscriptions
             put :content_override
+            get :content_override, :action => :product_content
```
Comment (daviddavis) [inline config/routes/api/v2.rb]: Why not call the action `list_content_overrides` or rename the route to `get :product_content`?

Comment (cfouant) [inline config/routes/api/v2.rb]: Okay - I wasn't a huge fan of how I enacted these methods in activation keys and was trying to match them while making it better. However, I'll change the route to get :product_content to keep it comparable to how it is already enabled in activation keys

**Comment by cfouant** [comment]
@daviddavis tests created, let me know what you think

**Comment by daviddavis** [comment]
The tests look good.

**Comment by daviddavis** [comment]
ACK

**Pull request merged**